src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -474,11 +474,19 @@ impl<'a,T:Payload> Ref<'a,T> {
             let remaining_ast_crumbs = ast_crumbs;
             Some(NodeFoundByAstCrumbs{node, ast_crumbs: remaining_ast_crumbs })
         } else {
-            let mut children = self.node.children.iter();
-            // Please be advised, that the `ch.ast_crumhs` is not a field of Ref, but Child, and
+            // Please be advised, that the `ch.ast_crumbs` is not a field of Ref, but Child, and
             // therefore have different meaning!
-            let next = children.find_position(|ch| {
+            let next = self.node.children.iter().find_position(|ch| {
                 !ch.ast_crumbs.is_empty() && ast_crumbs.starts_with(&ch.ast_crumbs)
+            }).or_else(|| {
+                // We do the second round to cover case of "prefix-like" nodes with
+                // `InsertionPoint(ExpectedArgument(_))`. See also docs for
+                // `generate::generate_expected_argument`.
+                // TODO[ao]: This is kinda heuristic. Should be reconsidered in
+                //  https://github.com/enso-org/ide/issues/787
```

Comment:
This comment is not understandable for anyone but the author. Could you elaborate more why this is heuristic, what could be better ways to do it etc, please?

Reply:
But there is explanation, just a line above?